### ISSUE TITLE
Testing notes: Rubbing on sweater

### ISSUE THREAD

**Issue by terracoda** [phetsims/balloons-and-static-electricity#323]
Phet test build
System: Mac OS Siera 10.12.3
Browser: Safari 10.0.3

http://phettest.colorado.edu/balloons-and-static-electricity/balloons-and-static-electricity_en.html?ea&brand=phet&accessibility

**Keyboard no VO**
- arrow and wasd keys work fine on own and with shift key
- all controls work fine
- [x] KNown issue: visual grab and move hint visual needs fixing (missing white space)

**Keyboard with VO**
- [x] Known issue: Focus lost when I use arrow keys to move a grabbed balloon with VO on. I'm not sure when the issue cropped up. I think it has been around since work on the new grid started.

**Location issues**
- [x] Incorrect preposition: On left edge. should be "At left edge."
- [x] Incorrect use of location: "On left edge" is not a position on the sweater. The edge is a boundary location only and should be "At left edge." Grab and release that happen "At left edge", should have a location somewhere along the "left arm of sweater" or "left shoulder". For example, on release, "Released. moved quickly right. Sticking to lower-left arm of sweater."  On grab, "Grabbed. On lower-left arm. Has..."

**Sweater special case**
- [x] Grab Alert: Sweater description incorrect when there are no more negative charges in Show all charges view only.

_Full example:_
Incorrect depleted sweater description in Show all charges view: 
"Grabbed. On upper-right side of sweater. Has many more negative charges than positive charges. Sweater has ALL MORE POSITIVE CHARGES THAN NEGATIVE CHARGES." 

_Correct special case depleted sweater description:_ 
"Grabbed. On upper-right side of sweater. Has many more negative charges than positive charges. Sweater has [[no more negative charges, only positive charges]]."

**Releases**
- [x] Releases off the sweater do not get attractive state and location, only location, "On sweater" at end of release. This will likely be fixed when you focus on the release alerts.

**Comment by jessegreenberg** [comment]
> Known issue: Focus lost when I use arrow keys to move a grabbed balloon with VO on. I'm not sure when the issue cropped up. I think it has been around since work on the new grid started.

@terracoda can you please verify that you have quick nav turned off?  I can't reproduce this behavior on MacOS 10.12.1 with Safari 10.0.1, but when quick nav is turned on the arrow keys should no longer work on the balloon.

**Comment by jessegreenberg** [comment]
> Nown issue: visual grab and move hint visual needs fixing (missing white space)

Checking this off, it is going to be tracked in #319.

**Comment by jessegreenberg** [comment]
> Releases off the sweater do not get attractive state and location, only location,

Working on this next.

**Comment by jessegreenberg** [comment]
With the above commits, release alerts are updated to match the latest design docs.

Thanks for the issue list @terracoda. https://github.com/phetsims/balloons-and-static-electricity/issues/323#issuecomment-342266216 is the last part of this issue.

**Comment by jessegreenberg** [comment]
The above commits also fixed a bug where some prepositions were getting duplicated ("At at center of play area" "Near near wall" and so on).

**Comment by terracoda** [comment]
@jessegreenberg re https://github.com/phetsims/balloons-and-static-electricity/issues/323#issuecomment-342266216, yes, sorry, quick nav was on. If I make sure quick nav is off, the arrow keys move the balloon.

**Comment by terracoda** [comment]
@jessegreenberg, great work.
I tested the sweater rubbing alerts today and they have changed, so I am creating a new issue for those and closing this one.

**Comment by terracoda** [comment]
The new testing notes are in this issue: https://github.com/phetsims/balloons-and-static-electricity/issues/324 .
